crates/router/src/core/payments.rs

```diff
@@ -3364,6 +3379,64 @@ where
     Ok(payment_data)
 }
 
+#[cfg(feature = "v1")]
+pub async fn get_session_token_for_click_to_pay(
+    state: &SessionState,
+    merchant_id: &id_type::MerchantId,
+    key_store: &domain::MerchantKeyStore,
+    authentication_product_ids: serde_json::Value,
+    payment_intent: &hyperswitch_domain_models::payments::PaymentIntent,
+) -> RouterResult<api_models::payments::SessionToken> {
+    use common_utils::id_type::MerchantConnectorAccountId;
+    use hyperswitch_domain_models::payments::ClickToPayMetaData;
+
+    use crate::consts::CLICK_TO_PAY;
+
+    let mca_ids: HashMap<String, MerchantConnectorAccountId> = authentication_product_ids
+        .parse_value("HashMap<String, MerchantConnectorAccountId>")
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Error while parsing authentication product ids")?;
+    let click_to_pay_mca_id = mca_ids
+        .get(CLICK_TO_PAY)
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Error while getting click_to_pay mca_id from business profile")?;
+    let key_manager_state = &(state).into();
+    let merchant_connector_account = state
+        .store
+        .find_by_merchant_connector_account_merchant_id_merchant_connector_id(
+            key_manager_state,
+            merchant_id,
+            click_to_pay_mca_id,
+            key_store,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
+            id: click_to_pay_mca_id.get_string_repr().to_string(),
+        })?;
+    let click_to_pay_metadata: ClickToPayMetaData = merchant_connector_account
```

Comment:
are we validating this metadata when it is being stored in the database?

Reply:
Yes, it's in the PR where the CTP_Mastercard connector is added